### DISCUSSION TITLE
ccache in VS2019 with Ninja failed rename error

### DISCUSSION THREAD

**Discussion by guchangx** [ccache/ccache#771] [Questions & answers]
error : failed to rename C:\Users\gucha\AppData\Roaming/ccache/tmp/tmp.cpp_stdout.P1kAlM to C:\Users\gucha\AppData\Roaming/ccache/tmp/tmp.cpp_stdout.P1kAlM.ii: ???????????

i do not find C:\Users\gucha\AppData\Roaming/ccache/tmp/tmp.cpp_stdout.P1kAlM in my local.

![image](https://user-images.githubusercontent.com/73569810/103964711-10480900-5197-11eb-974b-f46c9603eb92.png)

windows
VS2019

I tansplanted the ccache into windows

Does anyone know how to fix it.

**Comment by AlexanderLanin** [comment]
Can you provide more info on what you did? How to reproduce this? What version you used? etc

**Reply by guchangx** [reply]
I appreciate your reply to my message.

pc:  window 10 /x64
IDE: vs 2019 with ninja 
ccache version: i complie it from two weeks ago master code.
i add CMAKE_C_COMPILER_LAUNCHER to CMakelist.txt like that
![image](https://user-images.githubusercontent.com/73569810/104312700-ecc3ec00-5511-11eb-9f60-02b9e7860b7d.png)
i execute the build with ninja.
always show Error	
failed to rename C:\Users\gucha\AppData\Roaming/ccache/tmp/tmp.cpp_stdout.FzP5tM to C:\Users\gucha\AppData\Roaming/ccache/tmp/tmp.cpp_stdout.FzP5tM.ii: ????????????,???????	

i do have "../tmp.cpp_stdout.FzP5tM"  in my local.


This is complete output from ninja
  [41/1234] D:\WorkTool\ccache\ccache.exe D:\WorkTool\VS2019\Enterprise\VC\Tools\MSVC\14.27.29110\bin\Hostx64\x64\cl.exe  /nologo /TP -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x050500 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_URL_CAST_FROM_STRING -DQT_USE_FAST_CONCATENATION -DQT_USE_FAST_OPERATOR_PLUS -DUNICODE -D_UNICODE -D_USING_V110_SDK71_=1 -Dgammaray_common_EXPORTS -Icommon -I..\..\..\common -Icommon\gammaray_common_autogen\include -I..\..\..\ -I..\..\..\3rdparty -I. -ID:\WorkTool\Qt\5.15.1.2\out64\include -ID:\WorkTool\Qt\5.15.1.2\out64\include\QtCore -ID:\WorkTool\Qt\5.15.1.2\out64\.\mkspecs\win32-msvc -ID:\WorkTool\Qt\5.15.1.2\out64\include\QtNetwork -ID:\WorkTool\Qt\5.15.1.2\out64\include\QtGui -ID:\WorkTool\Qt\5.15.1.2\out64\include\QtANGLE /DWIN32 /D_WINDOWS /W3 /GR /EHsc /wd4244 /wd4267 /MDd /Zi /Ob0 /Od /RTC1 /showIncludes /Focommon\CMakeFiles\gammaray_common.dir\gammaray_common_autogen\mocs_compilation.cpp.obj /Fdcommon\CMakeFiles\gammaray_common.dir\ /FS -c common\gammaray_common_autogen\mocs_compilation.cpp
  FAILED: common/CMakeFiles/gammaray_common.dir/gammaray_common_autogen/mocs_compilation.cpp.obj 
  D:\WorkTool\ccache\ccache.exe D:\WorkTool\VS2019\Enterprise\VC\Tools\MSVC\14.27.29110\bin\Hostx64\x64\cl.exe  /nologo /TP -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x050500 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_URL_CAST_FROM_STRING -DQT_USE_FAST_CONCATENATION -DQT_USE_FAST_OPERATOR_PLUS -DUNICODE -D_UNICODE -D_USING_V110_SDK71_=1 -Dgammaray_common_EXPORTS -Icommon -I..\..\..\common -Icommon\gammaray_common_autogen\include -I..\..\..\ -I..\..\..\3rdparty -I. -ID:\WorkTool\Qt\5.15.1.2\out64\include -ID:\WorkTool\Qt\5.15.1.2\out64\include\QtCore -ID:\WorkTool\Qt\5.15.1.2\out64\.\mkspecs\win32-msvc -ID:\WorkTool\Qt\5.15.1.2\out64\include\QtNetwork -ID:\WorkTool\Qt\5.15.1.2\out64\include\QtGui -ID:\WorkTool\Qt\5.15.1.2\out64\include\QtANGLE /DWIN32 /D_WINDOWS /W3 /GR /EHsc /wd4244 /wd4267 /MDd /Zi /Ob0 /Od /RTC1 /showIncludes /Focommon\CMakeFiles\gammaray_common.dir\gammaray_common_autogen\mocs_compilation.cpp.obj /Fdcommon\CMakeFiles\gammaray_common.dir\ /FS -c common\gammaray_common_autogen\mocs_compilation.cpp
E:\TestFuture\GammaRay\GammaRay\out\build\x64-Debug\ccache : error : failed to rename C:\Users\gucha\AppData\Roaming/ccache/tmp/tmp.cpp_stdout.FzP5tM to C:\Users\gucha\AppData\Roaming/ccache/tmp/tmp.cpp_stdout.FzP5tM.ii: ????????????,???????

**Reply by jrosdahl** [reply]
This is most likely bug #703, fixed by #736 which will be part of ccache 4.2.

> ccache version: i complie it from two weeks ago master code.

Which revision more exactly? Did it include eead7e9d6399? If not, please try again with latest master.

**Reply by guchangx** [reply]
thank you very much,i will build latest master code again.thank you  for your efforts.
